### PR TITLE
Add option for last date limit to PCRData.positive_rate() + simplify colored maps legend

### PR DESCRIPTION
**Pull request by Inglezos**
## Related issues
This is related to issues #745 and #747.

## What was changed
1. Added argument `last_date` in `PCRData.positive_rate()`.
2. Use horizontal legend in colored maps

Call example for `PCRData.positive_rate()`:
```
import covsirphy as cs
data_loader = cs.DataLoader()
pcr_data = data_loader.pcr()
country = "Italy"
pcr_data.positive_rate(country, last_date="21Apr2021")

```

### CODE REVIEW

File: covsirphy/cleaning/pcr_data.py [hunk 1]
```diff
@@ -571,6 +572,9 @@ def positive_rate(self, country, province=None, window=7, show_figure=True, file
         except PCRIncorrectPreconditionError:
             raise PCRIncorrectPreconditionError(
                 country=country, province=province, message="Too many missing Tests records") from None
+        # Limit tests records to last date
+        if last_date is not None:
+            subset_df = subset_df.loc[subset_df[self.DATE] <= last_date]
```
Comment (lisphilar) [inline covsirphy/cleaning/pcr_data.py:577]: Thank you for your issues and pull requests!
Please update the test of tests/test_cleaning/test_pcr.py::TestPCRData::test_positive_rate to check `last_date` argument using `@pytest.mark.parametrize`.

In line 577, I think we should replace `last_date` with `pd.to_datetime(last_date)`.

Comment (Inglezos) [reply covsirphy/cleaning/pcr_data.py]: For some reason it works too with just `<= last_date`. I will update it though.

Comment (lisphilar) [reply covsirphy/cleaning/pcr_data.py]: Oops, I tried it and confirmed it works. Sorry for that.

Comment (Inglezos) [reply covsirphy/cleaning/pcr_data.py]: It is better practice to show purpose of usage, I kept `pd.to_datetime()`, thank you.

**Review by lisphilar — APPROVED** [review]
Thank you for your pull request and updating!
This will be merged after completion of tests.
